pkg/services/sqlstore/team.go

```diff
@@ -204,12 +196,8 @@ func (ss *SQLStore) SearchTeams(ctx context.Context, query *models.SearchTeamsQu
 		params = append(params, user)
 	}
 
-	if query.UserIdFilter == models.FilterIgnoreUser {
-		sql.WriteString(getTeamSelectSQLBase(filteredUsers))
-	} else {
-		sql.WriteString(getTeamSelectWithPermissionsSQLBase(filteredUsers))
-		params = append(params, query.UserIdFilter)
-	}
+	sql.WriteString(getTeamSelectSQLBase(filteredUsers))
+	params = append(params, query.UserIdFilter)
```

Comment:
we could potentially use query.SignedInUser.UserID here so that the permission column will be correct for admins, as right now it won't work for them because query.UserIdFilter will be 0 to disable filtering.